ReactNativeClient/lib/reducer.js

```diff
@@ -362,9 +528,9 @@ const reducer = (state = defaultState, action) => {
 		case 'NOTE_SELECT_ADD':
 		case 'NOTE_SELECT_REMOVE':
 		case 'NOTE_SELECT_TOGGLE':
-			newState = changeSelectedNotes(state, action);
+			newState = handleHistory(newState, action);
+			newState = changeSelectedNotes(newState, action);
```

Comment:
I'm not happy with this. I think the better way to do this would be to make another history reducer that also listens to events like "NOTE_SELECT", "NOTE_DELTE" etc and does the appropriate things to maintain history.

Then we can combine them both using combineReducers.